src/node/waku_node.py

```diff
@@ -74,15 +77,65 @@ def start(self, **kwargs):
             key = key.replace("_", "-")
             default_args[key] = value
 
-        self._container = self._docker_manager.start_container(self._docker_manager.image, self._ports, default_args, self._log_path, self._ext_ip)
-        logger.debug(f"Started container from image {self._image_name}. REST: {self._rest_port}")
-        DS.waku_nodes.append(self)
-        delay(1)  # if we fire requests to soon after starting the node will sometimes fail to start correctly
-        try:
-            self.ensure_ready()
-        except Exception as ex:
-            logger.error(f"REST service did not become ready in time: {ex}")
-            raise
+        rln_args = {}
+        rln_register_only = default_args["rln-register-only"]
+        rln_credentials_set = True
+
+        if len(default_args["rln-creds"]) != 5 or any(value is None for value in default_args["rln-creds"].values()):
+            rln_credentials_set = False
+            logger.info(f"RLN credentials not set, starting without RLN")
+
+        if rln_credentials_set:
+            if rln_register_only:
+                if self.is_gowaku():
+                    rln_args["generate-rln-credentials"] = None
+
+                elif self.is_nwaku():
+                    rln_args["generateRlnKeystore"] = None
+                    rln_args["--execute"] = None
+            else:
+                rln_args["rln-relay"] = "true"
+
+            if self.is_gowaku():
+                self._volumes.extend(["/go-waku_rln_tree:/etc/rln_tree", "/go-waku_keystore:/keystore"])
+                rln_args["eth-account-private-key"] = default_args["rln-creds"]["go_waku_eth_client_private_key"]
+
+            elif self.is_nwaku():
+                self._volumes.extend(["/nwaku_rln_tree:/etc/rln_tree", "/nwaku_keystore:/keystore"])
+                rln_args["rln-relay-eth-private-key"] = default_args["rln-creds"]["nwaku_eth_client_private_key"]
+
+            rln_args.update(
+                {
+                    "rln-relay-cred-path": "/keystore/keystore.json",
+                    "rln-relay-cred-password": default_args["rln-creds"]["keystore_password"],
+                    "rln-relay-eth-client-address": default_args["rln-creds"]["eth_client_address"],
+                    "rln-relay-eth-contract-address": default_args["rln-creds"]["eth_contract_address"],
+                }
+            )
+
+            default_args.update(rln_args)
+
+        del default_args["rln-creds"]
+        del default_args["rln-register-only"]
+
+        if rln_register_only:
+            self._container = self._docker_manager.start_container(
```

Comment:
Container itself acts as a single command during RLN registration. It either finish with a success or an error. It may hang due to some connection issues. Not sure if we need to check this? Otherwise no need to stop.